Add private channel obfuscation docs (#8454)
discord/discord-api-docs: developers/topics/permissions.mdx

There may be other cases in which certain permissions implicitly deny or allow other permissions. In all cases, it is based on logical conclusions about how a user with certain permissions should or should not interact with Discord.

## Channel Visibility

> Beyond the permission calculations above, `VIEW_CHANNEL` also governs what an app can see about a channel. Over the Gateway, channels an app lacks `VIEW_CHANNEL` on are dispatched with obfuscated data if the app has opted into the relevant capability. Starting October 12, 2026, [Get Guild Channels](/developers/resources/guild#get-guild-channels) will also omit those channels from its response entirely. See [Obfuscated Channels](/developers/resources/channel#channel-object-obfuscated-channels) for information about how to handle them, and how to detect and surface hidden channels to server admins.

Reviewer: This seems to indicate you can escape obfuscation indefinitely by not providing the capability

> Beyond the permission calculations above, `VIEW_CHANNEL` also governs what an app can see about a channel. Over the Gateway, channels an app lacks `VIEW_CHANNEL` on are dispatched with obfuscated data if the app has opted into the relevant capability. Starting October 12, 2026, [Get Guild Channels](/developers/resources/guild#get-guild-channels) will also omit those channels from its response entirely. See [Obfuscated Channels](/developers/resources/channel#channel-object-obfuscated-channels) for information about how to handle them, and how to detect and surface hidden channels to server admins.

Reviewer: terminology nits

```suggestion
Beyond the permission calculations above, `VIEW_CHANNEL` also governs what a bot can see about a channel. Over the Gateway, channels a bot lacks `VIEW_CHANNEL` on are dispatched with obfuscated data if the bot client has opted into the relevant capability. Starting October 12, 2026, [Get Guild Channels](/developers/resources/guild#get-guild-channels) will also omit those channels from its response entirely. See [Obfuscated Channels](/developers/resources/channel#channel-object-obfuscated-channels) for information about how to handle them, and how to detect and surface hidden channels to server admins.
```

> Beyond the permission calculations above, `VIEW_CHANNEL` also governs what a bot can see about a channel. Over the Gateway, channels a bot lacks `VIEW_CHANNEL` on are dispatched with obfuscated data. [Get Guild Channels](/developers/resources/guild#get-guild-channels) will also omit those channels from its response entirely. This is currently true if the client has opted into the `CHANNEL_OBFUSCATION` [capability](/developers/events/gateway-events#identify-gateway-capabilities). Starting October 12, 2026, it will be true for all bots. See [Obfuscated Channels](/developers/resources/channel#channel-object-obfuscated-channels) for information.

Reviewer: Couple things here:
- I made a comment above adjusting some of the language around "permission to view a channel" (including how categories work), I wonder if theres a centralized place to include this that we can link both to (or just link one to the other?)
- The framing here makes it sound like you can opt into omitting channels from `get_guild_channels`, which isnt the case

There may be other cases in which certain permissions implicitly deny or allow other permissions. In all cases, it is based on logical conclusions about how a user with certain permissions should or should not interact with Discord.

## Channel Visibility

> Over the [Gateway](/developers/events/gateway), a client will receive a channel object with obfuscated metadata for guild channels it does not have permission to view. [Get Guild Channels](/developers/resources/guild#get-guild-channels) will also omit those channels from its response entirely. Permission to view a channel is defined by having `VIEW_CHANNEL` permission on a channel or by being connected to a voice channel. Channel categories are viewable if any of their children channels are viewable.

Reviewer 2: ```suggestion
Over the [Gateway](/developers/events/gateway), a client will receive a channel object with obfuscated metadata for guild channels it does not have permission to view. [Get Guild Channels](/developers/resources/guild#get-guild-channels) will also omit those channels from its response entirely. Permission to view a channel is defined by having `VIEW_CHANNEL` permission on a channel or by being connected to a voice channel. Channel categories are viewable if any of their child channels are viewable.
```

> Over the [Gateway](/developers/events/gateway), a client will receive a channel object with obfuscated metadata for guild channels it does not have permission to view. [Get Guild Channels](/developers/resources/guild#get-guild-channels) will also omit those channels from its response entirely. Permission to view a channel is defined by having `VIEW_CHANNEL` permission on a channel or by being connected to a voice a channel. Channel categories are viewable if any of their children channels are viewable.

Reviewer 2: ```suggestion
Over the [Gateway](/developers/events/gateway), a client will receive a channel object with obfuscated metadata for guild channels it does not have permission to view. [Get Guild Channels](/developers/resources/guild#get-guild-channels) will also omit those channels from its response entirely. Permission to view a channel is defined by having `VIEW_CHANNEL` permission on a channel or by being connected to a voice channel. Channel categories are viewable if any of their children channels are viewable.
```
